### ISSUE TITLE
fire elemental shaman issue

### ISSUE THREAD

**Issue by gandalkk** [GameFreedomGG/Sindragosa#459]
Decription:
fire elemental is broke hit for 13k should be like 3k or something
How it works:
i think it does too much damage my fire ele does close to 300k-400k damage if i snapshot it correctly in raids. possibly something wrong with the counting of ability damage? 
How it should work:
These are the formulas,
Swing: 180 + 0.6 * SP + ( 2.0 / 14 ) * AP                 /        **Theres noway this equals 13k with my stats** /
Blast: ( 725 to 825 ) + 0.2 * SP
Nova: ( 925-1050 ) + 0.5 * SP
Shield: 96 + 0.015 * SP
Base Mana: 4000 + 15(0.3 * Int
Source (you should point out proofs of your report, please give us some source):
https://github.com/VirginTeam/WotlkCore/issues/1322 has some retail log numbers from icc

[17:48:24.064] Greater Fire Elemental casts Fire Blast on Deathbringer Saurfang
[17:48:24.224] Greater Fire Elemental Fire Blast Deathbringer Saurfang 1927 (R: 214)
[17:48:24.623] Kåtsûmî gains Heroism from Kåtsûmî
[17:48:24.623] Kåtsûmî casts Heroism
[17:48:24.880] Apko gains Heroism from Kåtsûmî
[17:48:24.880] Kuznam gains Heroism from Kåtsûmî
[17:48:24.880] Dabears gains Heroism from Kåtsûmî
[17:48:24.880] Payne gains Heroism from Kåtsûmî
[17:48:24.880] Owld gains Heroism from Kåtsûmî
[17:48:24.880] Oranoss gains Heroism from Kåtsûmî
[17:48:24.880] Artzie gains Heroism from Kåtsûmî
[17:48:24.880] Shadowfiend gains Heroism from Kåtsûmî
[17:48:24.894] Irsupersdw gains Heroism from Kåtsûmî
[17:48:24.894] Anathime gains Heroism from Kåtsûmî
[17:48:24.894] Rootbasher gains Heroism from Kåtsûmî
[17:48:24.894] Gatina gains Heroism from Kåtsûmî
[17:48:24.894] Safari gains Heroism from Kåtsûmî
[17:48:24.894] Tntmaxy gains Heroism from Kåtsûmî
[17:48:24.894] Nosfe gains Heroism from Kåtsûmî
[17:48:24.894] Mehbee gains Heroism from Kåtsûmî
[17:48:24.969] Phlaryu gains Heroism from Kåtsûmî
[17:48:24.969] Razielz gains Heroism from Kåtsûmî
[17:48:24.969] Equity gains Heroism from Kåtsûmî
[17:48:24.969] Shadowfiend gains Heroism from Kåtsûmî
[17:48:24.972] Rogerbrown gains Heroism from Kåtsûmî
[17:48:24.972] Bakachu gains Heroism from Kåtsûmî
[17:48:24.972] Xerofour gains Heroism from Kåtsûmî
[17:48:24.972] Kyries gains Heroism from Kåtsûmî
[17:48:24.972] Dreamer gains Heroism from Kåtsûmî
[17:48:24.972] Tók gains Heroism from Kåtsûmî
[17:48:24.972] Thoothun gains Heroism from Kåtsûmî
[17:48:24.972] Triste gains Heroism from Kåtsûmî
[17:48:24.972] Blacadder gains Heroism from Kåtsûmî
[17:48:25.245] Greater Fire Elemental begins to cast Fire Nova
[17:48:27.687] Greater Fire Elemental Fire Nova Deathbringer Saurfang 4175 (R: 411)
[17:48:29.361] Greater Fire Elemental hits Deathbringer Saurfang 4268
[17:48:30.123] Greater Fire Elemental Fire Shield Deathbringer Saurfang 331 (R: 22)
[17:48:31.384] Greater Fire Elemental hits Deathbringer Saurfang 2939 (R: 377)

**Comment by styxel** [comment]
this was already reported but you got some extra info i guess to fix it

**Comment by demetrzz** [comment]
All the reports are "i think its bugged", "im pretty sure its bugged", "omg its broken, fix it!" etc, without any actual testing and just copypasting formulas from unknown source which all makes harder for devs to find whats actually bugged. Lets assume those formulas are correct and compare to actual numbers.

My stats, i rounded all numbers for convenience:

4300 AP
2500 SP

This is average data from 4 pulls (doing more seemed unreasonable since values are very similar) on target dummy without any buffs or debuffs, i didn't snapshot any trinkets or enchants, overall dmg from pulls was 212k on average during 2 minutes.

I added base damage to formulas from those sources:
https://wotlkdb.com/?npc=15438 - swing
https://wotlkdb.com/?spell=61655 - fire nova
https://wotlkdb.com/?spell=57984 - fire blast

swing: 180 + 0.6 * 2500 + (2 / 14) * 4300 + 160 = 2450
from tests: 3675

fire nova: 990 + 0.5 * 2500 + 160 = 2400
from tests: 2400

fire blast: 775 + 0.2 * 2500 + 120 = 1395
from tests: 1350

Didnt bother comparing damage from shield since dmg is too low either way.

We can see that **_fire nova and fire blast damage is actually correct_**, but **_swing is exactly 50% higher of what it should be assuming formula is correct._** So there is more to the formula that scales swing damage.

So in conclusion: **swing does 50% higher damage**, but if those formulas are correct and this damage is blizzlike or a bug on the server it needs to be checked by devs or possibly will require more datamining from retail forums and other sources, even rawr/enhsim/simcraft have only appoximate values for fire elemental afaik.

**Comment by gandalkk** [comment]
So on dummy u were hitting 3.7k ish with swings? How come its 13k+ in raids

**Comment by demetrzz** [comment]
1) 13k+ doesnt explain anything, show some data, screenshot your SP and AP and screenshot details report for fire ele melee attacks afterwards, and if the data doesnt lign up with formulas that will prove that it actually scales wrong. But i expect it to be 50% higher, since current actual formula from my tests looks like this:

(180 + 0.6 * 2500 + (2 / 14) * 4300 + 160) + 50% = 3675

So lets assume this formula is right and scaling is linear, with full raid buffs and all snapshots your stats would be 50% higher which is realistic: 

(180 + 0.6 * 3750 + (2 / 14) * 6450 + 160) + 50% = 5300

So if crit is 2x damage it would be 10600.

On top of that since the damage type is fire it should scale with all magic debuffs, like curse of elements, so here we might get your 13000 damage that you noticed somewhere.

2) Guardians like fire elemental shapshotting summoner's stats on summon, while normal pets like hunter's pets and feral wolves update stats dynamically. Guardians were changed to scale dynamically in patch 5.0.

3) If you go by that data from videos its actually proving that those formulas are wrong since normal hit in 4.6k gs gear should be 2400 (im spellhance also, so thats additional 580 spell damage from weapon) and for fire ele to hit for 1300 on average you need to be naked or something while people on the videos are obviously not.

4) There is no proof that Hellscream's Warsong or other % buffs dynamically affect guardians, it doesnt snapshot %dmg buffs, only SP and AP, so damage that it gets from snapshotted AP and SP might be final, but i didn't research it and also its irrelevant to the bug report here.

So in the end devs just need to check from where this 50% damage comes and if its intended to work like that or a bug imo.

**Comment by gandalkk** [comment]
Just look at all the retail videos it never hits over 5k any where even in icc gear should be enough to proof its doing way too much damage and the formula is probably not correct idk where its from anyway.

**Comment by demetrzz** [comment]
Yea those logs are kind of useless, have to check by details/skada, impossible to separate damage here, can see only highest crits i guess.

13000 again seems reasonable with this formula:

(180 + 0.6 * SP + (2 / 14) * AP + 160) + 50%

You get 13% more from CoE and 3% from ret pala debuff, also for hodir's fire, it is a debuff not a buff, so thats 60% on top of that.

Even if they remove this 50% scaling, fire elemental would be superior over magma totem anyway if you snapshot it correctly, only damage from melee swings would be less, like instead of 200k it would do 133k which is still a lot. And fire blast and fire nova damage seems correct, so cleaving would still be way better also than magma totem, it also gives you free globals obviously.

But thats not the point of this bug report anyway, we just need for devs to look from where this extra 50% dmg comes and if it's correct and thats it.

**Comment by styxel** [comment]
bro its not 50% stop saying its 50% :D. i mean retail videos are usually the best sources for proof how things should work ( if u want it to be blizzlike) and its very clear that its not 50% 

Fire ele has never been this powerful on any server ive played on. Fire ele was doing upto like 100k dmg /summon total before the initial fix in naxx which was probably blizzlike i dont know what was reported and why to make it like this but its retarted.

**Comment by gandalkk** [comment]
Well the fix was to increase fire eles damage by 300% ish, which im sure was not what they tried to do. But all the shaman are enjoying it too much while trying to say its right :)

**Comment by demetrzz** [comment]
Why are you even arguing that its not 50%, my tests literally show it, you can see my numbers, do the math yourself if you want. Im using same formulas that are being posted around all server's githubs.

Those formulas are all what we have, blizzlike they are or not.

"Fire ele has never been this powerful on any server ive played on." This adds nothing of value to the report, it could be bugged everywhere since it seems like every other server uses same formulas (from github bug reports).

Dont see anyone on those videos from retail snapshot fire ele with wild magic potion or trinket/enchant procs either. If its not 50%, then where is the issue? I see damage values from those videos, but without players showing their stats before they summon fire ele you can't datamine anything from it.

What is initial fix? Even on ptr here spellhance shamans were blasting top 1 dps on patchwerk, im assuming because of the fire ele.

So your comment provided nothing of value to this bug report. 

Instead of just whining that's its bugged like everyone in other bug reports go do some research and show some numbers, if you find video where people show stats before snapshot it will prove if formulas are correct or not or otherwise devs will mark this report as "needs testing" and will have nothing to work with, so it will stay like that forever.

**Comment by styxel** [comment]
u could test it without procs if its smacks for 1.6k or more like the icc shaman does on that video

https://github.com/GameFreedomGG/Sindragosa/issues/40

**Comment by demetrzz** [comment]
> u could test it without procs if its smacks for 1.6k or more like the icc shaman does on that video
> 
> #40
> 
> https://youtu.be/j4s9xilLlNU?t=370 u can actually see here he snapshots it with atleast with skull trinket from icc and fully stacked frost trinket again i have no experience if it snapshots from attackpower just putting this here just saw this report and i agree about fire ele.
> 
> could try to find more videos later..

"u could test it without procs if its smacks for 1.6k" I guess you didn't even read my first post, go ahead and do it carefully.

You still dont understand how those videos dont help fixing this bug?

**Comment by gandalkk** [comment]
Wtf is wrong with your head im done here anyway i think there enough proofs here have a nice day

**Comment by styxel** [comment]
All we have is videos from retail and this formula which i dont know where it origins from. No need to get aggressive xD

**Comment by demetrzz** [comment]
Well he was the one who posted those formulas without any data or tests. I showed that difference from those formulas is exactly 50% and then he shifted his opinion on his formulas to "_idk where its from anyway"_ and started throwing random numbers like "300% ish" without actually suggesting any ways to fix this bug or providing any data that could help with that.

After that he in passive-agressive manner said that im trying to prevent this bug from getting fixed which makes no sense since im the one who put the effort in and now from his last comment you see that he got mad on himself for not being educated enough i guess to understand simple math and why his bug report, videos and comments not helping with fixing anything.

And its not just him, all the other bug reports and comments are just people who want this bug to be fixed but they are just whining about damage being broken while contributing nothing, and thats the main reason why nothing is being done about those bug reports, devs simply have nothing to work with.

> All we have is videos from retail and this formula which i dont know where it origins from. No need to get aggressive xD

100% agree, but you too for some reason think "its very clear not 50%", it's not like i disagree with you, there is no way atm to confirm if you right or wrong, but maybe you can figure out then how it's exactly supposed to work so devs can fix it? We just need someone who's willing to spend time to research and confirm that those formulas are correct or wrong and suggest other formulas based on players stats and swing damage from videos and logs.

**Comment by styxel** [comment]
They are wrong judging from damage of retail videos imo. I dont know, maybe it was broken in retail too and these formula are correct. I personally dont really care but if someone has the time maybe they can find something. I just said its not bugged by 50% when comparing to retail behaviour if it hits 6-14k here and on retail 4k max i have seen so far?
